bittide-instances/src/Bittide/Instances/Hitl/Transceivers.hs

```diff
@@ -28,20 +30,53 @@ import Clash.Explicit.Prelude
 import Bittide.Arithmetic.Time
 import Bittide.ClockControl.Si5395J
 import Bittide.ClockControl.Si539xSpi
-import Bittide.Hitl (HitlTests, hitlVioBool, allFpgas, noConfigTest)
+import Bittide.ElasticBuffer (sticky)
+import Bittide.Hitl (HitlTests, NoPostProcData(..), FpgaIndex, hitlVio)
 import Bittide.Instances.Domains
 import Bittide.Instances.Hitl.Setup
 import Bittide.Transceiver
 
+import Clash.Annotations.TH (makeTopEntity)
 import Clash.Cores.Xilinx.GTH
-import Clash.Cores.Xilinx.Xpm.Cdc.Single
+import Clash.Cores.Xilinx.Xpm.Cdc.Single (xpmCdcSingle)
 import Clash.Xilinx.ClockGen
+import Data.Maybe (fromMaybe, isJust)
 
+import qualified Bittide.Transceiver.ResetManager as ResetManager
 import qualified Clash.Explicit.Prelude as E
+import qualified Data.List as L
+import qualified Data.Map as Map
+import qualified Data.Text as Text
+
+-- | Start value of the counters used in 'counter' and 'expectCounter'. This is
+-- a non-zero start value, as a regression test for a bug where the transceivers
+-- would not come up if the counters started at zero.
+counterStart :: BitVector 64
+counterStart = 0xDEAD_BEEF_CA55_E77E
+
+-- | A counter starting at 'counterStart'
+counter :: KnownDomain dom => Clock dom -> Reset dom -> Signal dom Bool -> Signal dom (BitVector 64)
+counter clk rst ena = let c = register clk rst (toEnable ena) counterStart (c + 1) in c
+
+-- | Expect a counter starting at 'counterStart' and incrementing by one on each
+-- cycle.
+expectCounter ::
+  KnownDomain dom =>
+  Clock dom ->
+  Reset dom ->
+  Signal dom (Maybe (BitVector 64)) ->
+  -- ^ Received data
+  Signal dom Bool
+  -- ^ Error
+expectCounter clk rst = sticky clk rst . mealy clk rst enableGen go counterStart
+ where
+  go c (Just e) = (c + 1, c /= e)
+  go c Nothing  = (c, False)
 
 -- | Worker function for 'transceiversUpTest'. See module documentation for more
 -- information.
 goTransceiversUpTest ::
+  Signal Basic125 FpgaIndex ->
```

Comment:
Better use https://github.com/bittide/bittide-hardware/blob/703b881e8e468e7916dc01b2f8c49ba559ac4649/bittide-instances/src/Bittide/Instances/Hitl/Setup.hs#L22 instead of `FpgaIndex`, to avoid redundancies. The number of FPGAs in use should also not be fixed in `bittide-experiments` IMO.

Reply:
Though I agree, this is already part of `bittide-experiments`. Can we do this cleanup in another PR?

Reply:
That's fine for me. I added #539 to keep track of it.